### PR TITLE
seed ops.ssm_gp with some matern kernel utilities

### PR DESCRIPTION
**Pull request by martinjankowiak**
this is basically done but i still want to add support for nu=2.5

will have more demanding tests in the context of contrib.timeseries

### CODE REVIEW

File: pyro/ops/ssm_gp.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    def process_covariance(self, A):
+        """
+        Given a transition matrix `A` computed with `transition_matrix` compute the
+        the process covariance as described in Eqn. 3.11 in reference [2].
+
+        :returns torch.Tensor: a batched covariance matrix of shape (num_gps, state_dim, state_dim)
+        """
+        assert A.shape == (self.num_gps, self.state_dim, self.state_dim)
+        p = self.stationary_covariance()
+        q = p - torch.matmul(A.transpose(-1, -2), torch.matmul(p, A))
```
Comment (fritzo) [inline pyro/ops/ssm_gp.py]: I believe you can abbreviate
```py
q = p - A.transpose(-1, -2) @ p @ A
```

Comment (martinjankowiak) [reply pyro/ops/ssm_gp.py]: thanks but i really don't like using `@` 😬

**Comment by martinjankowiak** [comment]
i was going to include the log_prob tests in contrib.timeseries since otherwise i have to replicate a bunch of code to compute log_probs

**Comment by martinjankowiak** [comment]
@fritzo i fixed nu=2.5 so this is now ready for review. 

tests for exact correspondence between this and T^3 ops to follow elsewhere

(but actually the existing tests are already pretty demanding; a typo would probably cause something to be non PSD)

File: pyro/ops/ssm_gp.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+
+class MaternKernel(nn.Module):
+    """
+    Provides the building blocks for representing univariate Gaussian Processes (GPs)
+    with Matern kernels as state space models.
+
+    :param float nu: The order of the Matern kernel (1.5 or 2.5)
+    :param int num_gps: the number of GPs
+    :param torch.tensor log_length_scale_init: optional `num_gps`-dimensional vector of initializers
```
Comment (fritzo) [inline pyro/ops/ssm_gp.py]: `torch.tensor` -> `torch.Tensor`

File: pyro/ops/ssm_gp.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        for the length scale
+    :param torch.tensor log_kernel_scale_init: optional `num_gps`-dimensional vector of initializers
```
Comment (fritzo) [inline pyro/ops/ssm_gp.py]: `torch.tensor` -> `torch.Tensor`

File: pyro/ops/ssm_gp.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    [2] `Stochastic Differential Equation Methods for Spatio-Temporal Gaussian Process Regression`,
+        Arno Solin.
+    """
+    def __init__(self, nu=1.5, num_gps=1, log_length_scale_init=None, log_kernel_scale_init=None):
+        assert nu in [1.5, 2.5], "The only supported values of nu are 1.5 and 2.5"
+        self.nu = nu
+        self.state_dim = {1.5: 2, 2.5: 3}[nu]
+        self.num_gps = num_gps
+
+        if log_length_scale_init is not None:
```
Comment (fritzo) [inline pyro/ops/ssm_gp.py]: nit: in general it is cleaner to locate assertions after control flow so it is obvious that both code paths follow the same invariant. While you can indeed figure that out by reading the code, `sed` isn't as smart and often makes mistakes when editing code.
```py
if log_length_scale is None:
    log_length_scale_init = torch.zeros(num_gps)
assert log_length_scale_init.shape == (num_gps,)

if log_kernel_scale_init is None:
    log_kernel_scale_init = torch.zeros(num_gps)
assert log_kernel_scale_init.shape == (num_gps,)
```

File: pyro/ops/ssm_gp.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+    **References**
+
+    [1] `Kalman Filtering and Smoothing Solutions to Temporal Gaussian Process Regression Models`,
+        Jouni Hartikainen and Simo Sarkka.
+    [2] `Stochastic Differential Equation Methods for Spatio-Temporal Gaussian Process Regression`,
+        Arno Solin.
+    """
+    def __init__(self, nu=1.5, num_gps=1, log_length_scale_init=None, log_kernel_scale_init=None):
+        assert nu in [1.5, 2.5], "The only supported values of nu are 1.5 and 2.5"
```
Comment (fritzo) [inline pyro/ops/ssm_gp.py]: nit:
```py
if nu not in [1.5, 2.5]:
    raise NotImplementedError("The only supported values of nu are 1.5 and 2.5")
```

Comment (martinjankowiak) [reply pyro/ops/ssm_gp.py]: done

File: pyro/ops/ssm_gp.py (hunk 4 above)
Comment (martinjankowiak) [reply pyro/ops/ssm_gp.py]: done

Comment (martinjankowiak) [reply pyro/ops/ssm_gp.py]: done

File: pyro/ops/ssm_gp.py (hunk 3 above)
Comment (martinjankowiak) [reply pyro/ops/ssm_gp.py]: done

File: pyro/ops/ssm_gp.py (hunk 2 above)
Comment (martinjankowiak) [reply pyro/ops/ssm_gp.py]: done
